<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.32">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="author" content="Mathias">
<title>2&nbsp; Análisis genómico comparativo de genes DGF-1 en Trypanosoma Cruzi – Análisis de Secuencias</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./dinucleotideFreq.html" rel="next">
<link href="./index.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-fc6d358c97f25a8ea829b86655043430.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-1fe1bf4ce278c9b3058e1fb3636284ab.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script src="site_libs/core-js-2.5.3/shim.min.js"></script><script src="site_libs/react-18.2.0/react.min.js"></script><script src="site_libs/react-18.2.0/react-dom.min.js"></script><script src="site_libs/reactwidget-2.0.0/react-tools.js"></script><link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="site_libs/reactable-0.4.4/reactable.css" rel="stylesheet">
<script src="site_libs/reactable-binding-0.4.4/reactable.js"></script>
</head>
<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./geneFeature.html">Análisis Básico</a></li><li class="breadcrumb-item"><a href="./geneFeature.html"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Análisis genómico comparativo de genes DGF-1 en <em>Trypanosoma Cruzi</em></span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Análisis de Secuencias</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Introducción</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Análisis de Secuencias</span></span></a>
  </div>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">Análisis Básico</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./geneFeature.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Análisis genómico comparativo de genes DGF-1 en <em>Trypanosoma Cruzi</em></span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./dinucleotideFreq.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Nucleotide frequency</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./aminoAcidFreq.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Amino acid frequency</span></span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">Table of contents</h2>
   
  <ul>
<li>
<a href="#descripcion" id="toc-descripcion" class="nav-link active" data-scroll-target="#descripcion"><span class="header-section-number">2.1</span> Descripcion</a>
  <ul class="collapse">
<li><a href="#genomas" id="toc-genomas" class="nav-link" data-scroll-target="#genomas"><span class="header-section-number">2.1.1</span> Genomas</a></li>
  <li><a href="#librerias-y-datos-a-usar" id="toc-librerias-y-datos-a-usar" class="nav-link" data-scroll-target="#librerias-y-datos-a-usar"><span class="header-section-number">2.1.2</span> Librerias y datos a usar</a></li>
  </ul>
</li>
  <li>
<a href="#procesamiento-de-los-genomas" id="toc-procesamiento-de-los-genomas" class="nav-link" data-scroll-target="#procesamiento-de-los-genomas"><span class="header-section-number">2.2</span> Procesamiento de los genomas</a>
  <ul class="collapse">
<li><a href="#caracter%C3%ADsticas-de-los-genomas" id="toc-características-de-los-genomas" class="nav-link" data-scroll-target="#caracter%C3%ADsticas-de-los-genomas"><span class="header-section-number">2.2.1</span> Características de los genomas</a></li>
  <li><a href="#tabla-resumen-de-los-genomas" id="toc-tabla-resumen-de-los-genomas" class="nav-link" data-scroll-target="#tabla-resumen-de-los-genomas"><span class="header-section-number">2.2.2</span> Tabla resumen de los genomas</a></li>
  <li><a href="#visualizaci%C3%B3n-de-la-longitud-y-contenido-de-gc-de-los-genomas" id="toc-visualización-de-la-longitud-y-contenido-de-gc-de-los-genomas" class="nav-link" data-scroll-target="#visualizaci%C3%B3n-de-la-longitud-y-contenido-de-gc-de-los-genomas"><span class="header-section-number">2.2.3</span> Visualización de la Longitud y Contenido de GC de los Genomas</a></li>
  </ul>
</li>
  <li>
<a href="#procesamiento-del-total-de-las-secuencias-codificantes" id="toc-procesamiento-del-total-de-las-secuencias-codificantes" class="nav-link" data-scroll-target="#procesamiento-del-total-de-las-secuencias-codificantes"><span class="header-section-number">2.3</span> Procesamiento del total de las secuencias codificantes</a>
  <ul class="collapse">
<li><a href="#tabla-resumen-de-secuencias-codificantes-por-genoma" id="toc-tabla-resumen-de-secuencias-codificantes-por-genoma" class="nav-link" data-scroll-target="#tabla-resumen-de-secuencias-codificantes-por-genoma"><span class="header-section-number">2.3.1</span> Tabla resumen de secuencias codificantes por genoma</a></li>
  </ul>
</li>
  <li>
<a href="#procesamiento-de-las-secuencias-dgf-1" id="toc-procesamiento-de-las-secuencias-dgf-1" class="nav-link" data-scroll-target="#procesamiento-de-las-secuencias-dgf-1"><span class="header-section-number">2.4</span> Procesamiento de las secuencias <code>DGF-1</code></a>
  <ul class="collapse">
<li><a href="#visualizaci%C3%B3n-de-la-longitud-y-contenido-de-gc-de-dgf-1" id="toc-visualización-de-la-longitud-y-contenido-de-gc-de-dgf-1" class="nav-link" data-scroll-target="#visualizaci%C3%B3n-de-la-longitud-y-contenido-de-gc-de-dgf-1"><span class="header-section-number">2.4.1</span> Visualización de la Longitud y Contenido de GC de <code>DGF-1</code></a></li>
  </ul>
</li>
  <li>
<a href="#filtrado-de-las-secuencias-de-dgf-1-por-el-largo" id="toc-filtrado-de-las-secuencias-de-dgf-1-por-el-largo" class="nav-link" data-scroll-target="#filtrado-de-las-secuencias-de-dgf-1-por-el-largo"><span class="header-section-number">2.5</span> Filtrado de las secuencias de DGF-1 por el largo</a>
  <ul class="collapse">
<li><a href="#resumen-de-informaci%C3%B3n-de-dgf-1-completas-en-cada-genoma-filtradas-por-largo" id="toc-resumen-de-información-de-dgf-1-completas-en-cada-genoma-filtradas-por-largo" class="nav-link" data-scroll-target="#resumen-de-informaci%C3%B3n-de-dgf-1-completas-en-cada-genoma-filtradas-por-largo"><span class="header-section-number">2.5.1</span> Resumen de Información de DGF-1 completas en cada genoma (filtradas por largo)</a></li>
  <li><a href="#visualizaci%C3%B3n-de-la-longitud-y-contenido-de-gc-de-dgf-1-completas" id="toc-visualización-de-la-longitud-y-contenido-de-gc-de-dgf-1-completas" class="nav-link" data-scroll-target="#visualizaci%C3%B3n-de-la-longitud-y-contenido-de-gc-de-dgf-1-completas"><span class="header-section-number">2.5.2</span> Visualización de la Longitud y Contenido de GC de <code>DGF-1</code> completas</a></li>
  </ul>
</li>
  <li><a href="#resumen-de-informaci%C3%B3n-de-secuencias-anotadas-como-dgf-1-menores-al-largo-minimo-establecido" id="toc-resumen-de-información-de-secuencias-anotadas-como-dgf-1-menores-al-largo-minimo-establecido" class="nav-link" data-scroll-target="#resumen-de-informaci%C3%B3n-de-secuencias-anotadas-como-dgf-1-menores-al-largo-minimo-establecido"><span class="header-section-number">2.6</span> Resumen de Información de secuencias anotadas como <code>DGF-1</code> menores al largo minimo establecido</a></li>
  <li>
<a href="#caracterizacion-de-los-nuevos-genes-dgf-1-anotados" id="toc-caracterizacion-de-los-nuevos-genes-dgf-1-anotados" class="nav-link" data-scroll-target="#caracterizacion-de-los-nuevos-genes-dgf-1-anotados"><span class="header-section-number">2.7</span> Caracterizacion de los nuevos genes <code>DGF-1</code> anotados</a>
  <ul class="collapse">
<li><a href="#visualizaci%C3%B3n-de-la-longitud-y-contenido-de-gc-de-el-total-de-los-nuevos-genes-dedgf-1" id="toc-visualización-de-la-longitud-y-contenido-de-gc-de-el-total-de-los-nuevos-genes-dedgf-1" class="nav-link" data-scroll-target="#visualizaci%C3%B3n-de-la-longitud-y-contenido-de-gc-de-el-total-de-los-nuevos-genes-dedgf-1"><span class="header-section-number">2.7.1</span> Visualización de la Longitud y Contenido de GC de el total de los nuevos genes de<code>DGF-1</code></a></li>
  </ul>
</li>
  </ul></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./geneFeature.html">Análisis Básico</a></li><li class="breadcrumb-item"><a href="./geneFeature.html"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Análisis genómico comparativo de genes DGF-1 en <em>Trypanosoma Cruzi</em></span></a></li></ol></nav><div class="quarto-title">
<h1 class="title">
<span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Análisis genómico comparativo de genes DGF-1 en <em>Trypanosoma Cruzi</em></span>
</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Mathias </p>
          </div>
  </div>
    
  
    
  </div>
  


</header><div class="cell">
<style type="text/css">
p {
  text-align: justify
}

</style>
</div>
<section id="descripcion" class="level2" data-number="2.1"><h2 data-number="2.1" class="anchored" data-anchor-id="descripcion">
<span class="header-section-number">2.1</span> Descripcion</h2>
<p><img src="https://img.itch.zone/aW1nLzE3MTEyNjE5LmdpZg==/original/2%2BYVZg.gif" class="img-fluid" style="float:right;" alt="Illustration of a detective cat. Artwork by @https://shigemi.itch.io." width="301"></p>
<p><em>Trypanosoma cruzi</em>, el parásito causante de la enfermedad de Chagas, posee una familia de genes llamada <code>DGF-1</code> que juega un papel clave en su biología. En este estudio, realizamos un análisis comparativo de estos genes en distintos genomas para entender su diversidad y posibles funciones.</p>
<p>Los datos utilizados en este análisis fueron obtenidas a partir de diferentes genomas de <em>Trypanosoma cruzi</em>, disponibles en <a href="https://tritrypdb.org/tritrypdb/app"><strong>tritrydb</strong></a> y <a href="https://www.ncbi.nlm.nih.gov/"><strong>NCBI</strong></a>.</p>
<p>A partir de estos genomas, se extrajeron secuencias de los genes <code>DGF-1</code>, con el objetivo de realizar análisis comparativo. Inicialmente se estudió el contenido de <code>GC</code>, <code>longitud de las secuencias</code>, y otros aspectos relevantes para la caracterización de los genes de <code>DGF-1</code>.</p>
<section id="genomas" class="level3" data-number="2.1.1"><h3 data-number="2.1.1" class="anchored" data-anchor-id="genomas">
<span class="header-section-number">2.1.1</span> Genomas</h3>
<p>Este análisis utiliza los siguientes genomas <code>TriTrypDB-68_TcruziDm28c2018_Genome.fasta, TriTrypDB-68_TcruziBrazilA4_Genome.fasta, TriTrypDB-68_TcruziYC6_Genome.fasta, TcDm25_TcruziTcDm25H1_Genome.fasta</code>. A partir de los cuales se obtienen las secuencias para su estudio.</p>
</section><section id="librerias-y-datos-a-usar" class="level3" data-number="2.1.2"><h3 data-number="2.1.2" class="anchored" data-anchor-id="librerias-y-datos-a-usar">
<span class="header-section-number">2.1.2</span> Librerias y datos a usar</h3>
<p>Se cargan las <code>bibliotecas</code> necesarias para el análisis de datos genómicos, así como para la manipulación, transformación y visualización de dataframes, junto con la generación de gráficos en formato de arreglo. A continuación, se crea una lista de archivos que contiene los <code>genomas</code> a analizar, y se inicializa una <code>tabla</code> destinada a almacenar los resultados del análisis de los archivos FASTA.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Cargar las librerías necesarias</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://seqinr.r-forge.r-project.org/">seqinr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">gridExtra</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://glin.github.io/reactable/">reactable</a></span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<hr></section></section><section id="procesamiento-de-los-genomas" class="level2" data-number="2.2"><h2 data-number="2.2" class="anchored" data-anchor-id="procesamiento-de-los-genomas">
<span class="header-section-number">2.2</span> Procesamiento de los genomas</h2>
<p>Para procesar las secuencias FASTA, se crea una función que genera un dataframe vacío donde se almacenarán los elementos obtenidos a partir de una secuencia dada. Luego, se elimina la ruta del archivo para extraer el nombre del mismo. A continuación, se obtiene el identificador de cada una de las secuencias dentro del archivo FASTA, se calcula la longitud de la secuencia y el contenido de GC utilizando funciones de la biblioteca SEQINR.</p>
<section id="características-de-los-genomas" class="level3" data-number="2.2.1"><h3 data-number="2.2.1" class="anchored" data-anchor-id="características-de-los-genomas">
<span class="header-section-number">2.2.1</span> Características de los genomas</h3>
<p>Se utiliza ALGO la función mencionada para extraer diversas características de los genomas. Con estos datos, generamos la primera <code>Tabla resumen</code>, que agrupa los contigs de cada genoma <code>TriTrypDB-68_TcruziDm28c2018_Genome.fasta, TriTrypDB-68_TcruziBrazilA4_Genome.fasta, TriTrypDB-68_TcruziYC6_Genome.fasta, TcDm25_TcruziTcDm25H1_Genome.fasta</code>, mostrando el <code>número total de secuencias encontradas</code>, así como la <code>longitud</code> mínima, media y máxima, así como el contenido de <code>GC</code> mínimo, medio y máximo.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Load the table generated by the second script</span></span>
<span><span class="va">genomes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"genome_gc_length.tsv"</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># genomes &lt;- read.csv("../output_directory/genome_gc_length.csv", sep = ",", header = TRUE)</span></span>
<span></span>
<span></span>
<span><span class="co"># Crear una tabla con el conteo de secuencias por archivo</span></span>
<span><span class="va">genomes_summary</span> <span class="op">&lt;-</span> <span class="va">genomes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>Total_Sequences <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html">n</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>            min_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">Length</span><span class="op">)</span>,</span>
<span>            mean_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">Length</span><span class="op">)</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>            max_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">Length</span><span class="op">)</span>,</span>
<span>            min_gc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">GC_Content</span><span class="op">)</span>,</span>
<span>            mean_gc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">GC_Content</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>            max_gc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">GC_Content</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section><section id="tabla-resumen-de-los-genomas" class="level3" data-number="2.2.2"><h3 data-number="2.2.2" class="anchored" data-anchor-id="tabla-resumen-de-los-genomas">
<span class="header-section-number">2.2.2</span> Tabla resumen de los genomas</h3>
<div class="cell">
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-cd4c39fde66d13627d83" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-cd4c39fde66d13627d83">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"file":["TcruziBrazilA4","TcruziDm28c2018","TcruziTcDm25H1","TcruziYC6"],"Total_Sequences":[402,636,35,266],"min_length":[703,911,47166,678],"mean_length":[113325,83761,1105091,177512],"max_length":[2738928,1645565,2824530,2951016],"min_gc":[0.4151,0.3628,0.2425,0.4448],"mean_gc":[0.533,0.519,0.505,0.524],"max_gc":[0.6968,0.707,0.5395,0.5997]},"columns":[{"id":"file","name":"file","type":"character"},{"id":"Total_Sequences","name":"Total_Sequences","type":"numeric"},{"id":"min_length","name":"min_length","type":"numeric"},{"id":"mean_length","name":"mean_length","type":"numeric"},{"id":"max_length","name":"max_length","type":"numeric"},{"id":"min_gc","name":"min_gc","type":"numeric"},{"id":"mean_gc","name":"mean_gc","type":"numeric"},{"id":"max_gc","name":"max_gc","type":"numeric"}],"highlight":true,"dataKey":"dc44fe2530b046f8a19cbb3cd57126f8"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section><section id="visualización-de-la-longitud-y-contenido-de-gc-de-los-genomas" class="level3" data-number="2.2.3"><h3 data-number="2.2.3" class="anchored" data-anchor-id="visualización-de-la-longitud-y-contenido-de-gc-de-los-genomas">
<span class="header-section-number">2.2.3</span> Visualización de la Longitud y Contenido de GC de los Genomas</h3>
<p>En esta sección, se generarán gráficos que representan el <code>contenido de GC</code> y la <code>longitud</code> de cada genoma. Se implementan funciones específicas para cada gráfico, lo que permite reutilizar el código de manera eficiente en otras partes del análisis. Para crear estas funciones se implementara las bibliotecas <code>ggridges</code>, <code>ggbeeswarm</code> y <code>cowplot</code>.</p>
<div class="tabset-margin-container"></div><div class="panel-tabset nav-pills">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true" aria-current="page">GC</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false">Length</a></li>
</ul>
<div class="tab-content nav-pills">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<p>La función <code>plot_violin_gc</code> genera un gráfico de violín que muestra la distribución del contenido de GC en distintos genomas <code>TriTrypDB-68_TcruziDm28c2018_Genome.fasta, TriTrypDB-68_TcruziBrazilA4_Genome.fasta, TriTrypDB-68_TcruziYC6_Genome.fasta, TcDm25_TcruziTcDm25H1_Genome.fasta</code>.</p>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/ggridges/">ggridges</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/eclarke/ggbeeswarm">ggbeeswarm</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/">cowplot</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">genomes</span><span class="op">$</span><span class="va">Length</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/format.html">format</a></span><span class="op">(</span><span class="va">genomes</span><span class="op">$</span><span class="va">Length</span>, scientific <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="co"># Changed format</span></span>
<span></span>
<span><span class="co"># Inspired by http://tom-e-white.com/datavision/05-genome-size.html domain of https://github.com/tomwhite</span></span>
<span></span>
<span><span class="va">plot_violin_gc</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">data</span>, <span class="va">plot_title</span>, <span class="va">y_label</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># Sorting data</span></span>
<span>  <span class="va">data</span><span class="op">$</span><span class="va">file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">file</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">file</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">count_data</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>count <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html">n</a></span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">file</span>, y <span class="op">=</span> <span class="va">GC_Content</span>, fill <span class="op">=</span> <span class="va">file</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_violin.html">geom_violin</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.7</span>, width <span class="op">=</span> <span class="fl">0.5</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.1</span>, fill <span class="op">=</span> <span class="st">"white"</span>, alpha <span class="op">=</span> <span class="fl">0.3</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_fill_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">count_data</span>,</span>
<span>              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">file</span>, y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">GC_Content</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">*</span> <span class="fl">1.05</span>, </span>
<span>                  label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">count</span><span class="op">)</span><span class="op">)</span>,</span>
<span>              vjust <span class="op">=</span> <span class="fl">0</span>, color <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">3.5</span><span class="op">)</span> <span class="op">+</span></span>
<span>    </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span></span>
<span>      title <span class="op">=</span> <span class="va">plot_title</span>,</span>
<span>      y <span class="op">=</span> <span class="va">y_label</span><span class="co">#,</span></span>
<span>      <span class="co">#caption = "Data source: Facultad de Ciencias, Udelar"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://wilkelab.org/cowplot/reference/theme_minimal_grid.html">theme_minimal_vgrid</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span></span>
<span>      axis.line.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>      axis.ticks.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>      plot.margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span><span class="fl">5.5</span>, <span class="fl">5.5</span>, <span class="fl">5.5</span>, <span class="fl">5.5</span>, <span class="st">"pt"</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_y_continuous</a></span><span class="op">(</span>expand <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/expansion.html">expansion</a></span><span class="op">(</span>mult <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.05</span>, <span class="fl">0.15</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Function plot_violin_gc call</span></span>
<span><span class="fu">plot_violin_gc</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">genomes</span>, </span>
<span>  plot_title <span class="op">=</span> <span class="st">"Genome GC"</span>, </span>
<span>  y_label <span class="op">=</span> <span class="st">"GC Content"</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell quarto-layout-panel" data-layout="[100]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="geneFeature_files/figure-html/plotting-genome-gc-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Secuencias por genoma y contenido GC.</figcaption></figure>
</div>
</div>
</div>
</div>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<p>La función de R <code>plot_violin_length</code> crea un gráfico de violín que ilustra la distribución de las longitudes genómicas de los diferentes genomas <code>TriTrypDB-68_TcruziDm28c2018_Genome.fasta, TriTrypDB-68_TcruziBrazilA4_Genome.fasta, TriTrypDB-68_TcruziYC6_Genome.fasta, TcDm25_TcruziTcDm25H1_Genome.fasta</code>.</p>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">plot_violin_length</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">data</span>, <span class="va">type</span>, <span class="va">plot_title</span>, <span class="va">x_label</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">12345</span><span class="op">)</span></span>
<span>  <span class="co"># Sorting data</span></span>
<span>  <span class="va">data</span><span class="op">$</span><span class="va">file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">file</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">data</span><span class="op">$</span><span class="va">file</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>  <span class="va">plot_data</span> <span class="op">&lt;-</span> <span class="va">data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>count <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html">n</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>           median_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html">median</a></span><span class="op">(</span><span class="va">Length</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">ungroup</a></span><span class="op">(</span><span class="op">)</span> <span class="co">#%&gt;%</span></span>
<span>    <span class="co"># mutate(file = reorder(file, -median_length))</span></span>
<span>  </span>
<span>  <span class="va">label_data</span> <span class="op">&lt;-</span> <span class="va">plot_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="va">file</span>, .keep_all <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">plot_data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">Length</span>, <span class="va">file</span>, color <span class="op">=</span> <span class="va">file</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/ggbeeswarm/man/geom_quasirandom.html">geom_quasirandom</a></span><span class="op">(</span>groupOnX <span class="op">=</span> <span class="cn">FALSE</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span>, size <span class="op">=</span> <span class="fl">1</span>, dodge.width <span class="op">=</span> <span class="fl">0.9</span>, alpha <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span></span>
<span>      title <span class="op">=</span> <span class="va">plot_title</span>,</span>
<span>      x <span class="op">=</span> <span class="va">x_label</span><span class="co">#,</span></span>
<span>      <span class="co">#caption = "Data source: Facultad de Ciencias, Udelar"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://wilkelab.org/cowplot/reference/theme_minimal_grid.html">theme_minimal_vgrid</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span></span>
<span>      axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>      axis.line.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>      axis.ticks.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>      plot.margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span><span class="fl">5.5</span>, <span class="fl">40</span>, <span class="fl">5.5</span>, <span class="fl">5.5</span>, <span class="st">"pt"</span><span class="op">)</span></span>
<span>    <span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_color_brewer</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"Dark2"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">label_data</span>,</span>
<span>              <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="cn">Inf</span>, y <span class="op">=</span> <span class="va">file</span>, label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">count</span><span class="op">)</span><span class="op">)</span>,</span>
<span>              hjust <span class="op">=</span> <span class="op">-</span><span class="fl">0.2</span>, color <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">3.5</span>,</span>
<span>              position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_nudge.html">position_nudge</a></span><span class="op">(</span>y <span class="op">=</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html">coord_cartesian</a></span><span class="op">(</span>clip <span class="op">=</span> <span class="st">"off"</span><span class="op">)</span></span>
<span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="va">type</span> <span class="op">==</span> <span class="st">"genome"</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_log10</a></span><span class="op">(</span></span>
<span>      breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">10</span><span class="op">^</span><span class="fl">2</span>, <span class="fl">10</span><span class="op">^</span><span class="fl">3</span>, <span class="fl">10</span><span class="op">^</span><span class="fl">4</span>, <span class="fl">10</span><span class="op">^</span><span class="fl">5</span>, <span class="fl">10</span><span class="op">^</span><span class="fl">6</span>, <span class="fl">10</span><span class="op">^</span><span class="fl">7</span>, <span class="fl">10</span><span class="op">^</span><span class="fl">8</span>, <span class="fl">10</span><span class="op">^</span><span class="fl">9</span>, <span class="fl">10</span><span class="op">^</span><span class="fl">10</span><span class="op">)</span>,</span>
<span>      labels <span class="op">=</span> <span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/trans_format.html">trans_format</a></span><span class="op">(</span><span class="st">"log10"</span>, <span class="fu">scales</span><span class="fu">::</span><span class="fu"><a href="https://scales.r-lib.org/reference/parse_format.html">math_format</a></span><span class="op">(</span><span class="fl">10</span><span class="op">^</span><span class="va">.x</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>    <span class="va">p</span> <span class="op">&lt;-</span> <span class="va">p</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_continuous</a></span><span class="op">(</span>expand <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/expansion.html">expansion</a></span><span class="op">(</span>mult <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.05</span>, <span class="fl">0.2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">p</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># Function plot_violin_length call</span></span>
<span><span class="fu">plot_violin_length</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">genomes</span>,</span>
<span>  <span class="st">"genome"</span>, </span>
<span>  plot_title <span class="op">=</span> <span class="st">"Genome size"</span>, </span>
<span>  x_label <span class="op">=</span> <span class="st">"Number of base pairs"</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell quarto-layout-panel" data-layout="[100]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="geneFeature_files/figure-html/plotting-genome-length-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Secuencias por genoma y contenido GC.</figcaption></figure>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
<hr></section></section><section id="procesamiento-del-total-de-las-secuencias-codificantes" class="level2" data-number="2.3"><h2 data-number="2.3" class="anchored" data-anchor-id="procesamiento-del-total-de-las-secuencias-codificantes">
<span class="header-section-number">2.3</span> Procesamiento del total de las secuencias codificantes</h2>
<p>Se llama a la funcion <code>calculate_fasta_feature</code> para calcular en el total de <code>secuencia codificante</code> el contenido de <code>GC</code> y la <code>longitud (pb)</code>. Finalmente, creamos una <code>tabla resumen</code>, que agrupa el <code>total secuencias codificantes</code> por cada genoma <code>TriTrypDB-68_TcruziDm28c2018_Genome.fasta, TriTrypDB-68_TcruziBrazilA4_Genome.fasta, TriTrypDB-68_TcruziYC6_Genome.fasta, TcDm25_TcruziTcDm25H1_Genome.fasta</code>, mostrando así como la <code>longitud</code> mínima, media y máxima, así como el contenido de <code>GC</code> mínimo, medio y máximo.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Load the table generated by the second script</span></span>
<span><span class="va">all_code_prot_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"all_code_protein_gc_length.tsv"</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># all_code_prot_data &lt;- read.csv("../output_directory/all_code_protein_gc_length.csv", sep = ",", header = TRUE)</span></span>
<span></span>
<span><span class="co"># Crear una tabla con el conteo de secuencias por archivo</span></span>
<span><span class="va">all_code_prot_summary</span> <span class="op">&lt;-</span> <span class="va">all_code_prot_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>Total_Sequences <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html">n</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>            min_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">Length</span><span class="op">)</span>,</span>
<span>            mean_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">Length</span><span class="op">)</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>            max_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">Length</span><span class="op">)</span>,</span>
<span>            min_gc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">GC_Content</span><span class="op">)</span>,</span>
<span>            mean_gc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">GC_Content</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>            max_gc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">GC_Content</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="tabla-resumen-de-secuencias-codificantes-por-genoma" class="level3" data-number="2.3.1"><h3 data-number="2.3.1" class="anchored" data-anchor-id="tabla-resumen-de-secuencias-codificantes-por-genoma">
<span class="header-section-number">2.3.1</span> Tabla resumen de secuencias codificantes por genoma</h3>
<div class="cell">
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-54c85ca0228bfe34e300" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-54c85ca0228bfe34e300">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"file":["TcruziBrazilA4","TcruziDm28c2018","TcruziTcDm25H1","TcruziYC6"],"Total_Sequences":[7990,7806,6005,7107],"min_length":[102,156,"NA",150],"mean_length":[1215,1453,"NA",1504],"max_length":[19908,15444,"NA",19011],"min_gc":[0.2767,0.3354,"NA",0.347],"mean_gc":[0.545,0.532,"NA",0.538],"max_gc":[0.7513,0.7069,"NA",0.7149]},"columns":[{"id":"file","name":"file","type":"character"},{"id":"Total_Sequences","name":"Total_Sequences","type":"numeric"},{"id":"min_length","name":"min_length","type":"numeric"},{"id":"mean_length","name":"mean_length","type":"numeric"},{"id":"max_length","name":"max_length","type":"numeric"},{"id":"min_gc","name":"min_gc","type":"numeric"},{"id":"mean_gc","name":"mean_gc","type":"numeric"},{"id":"max_gc","name":"max_gc","type":"numeric"}],"highlight":true,"dataKey":"47cbd4f408fba3835b4966a3dc0891a8"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="tabset-margin-container"></div><div class="panel-tabset nav-pills">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-2-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-1" role="tab" aria-controls="tabset-2-1" aria-selected="true">GC</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-2" role="tab" aria-controls="tabset-2-2" aria-selected="false">Length</a></li>
</ul>
<div class="tab-content nav-pills">
<div id="tabset-2-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-2-1-tab">
<p>Llamamos a la funcion <code>plot_violin_gc</code>, dando lugar a un gráfico violín que muestra la distribución del contenido de GC en distintos genomas de las secuencias.</p>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Function plot_violin_gc call</span></span>
<span><span class="fu">plot_violin_gc</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">all_code_prot_data</span>, </span>
<span>  plot_title <span class="op">=</span> <span class="st">"Coding sequence GC"</span>, </span>
<span>  y_label <span class="op">=</span> <span class="st">"GC Content"</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell quarto-layout-panel" data-layout="[100]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="geneFeature_files/figure-html/plotting-gc all coding protein-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Secuencias por genoma y contenido GC.</figcaption></figure>
</div>
</div>
</div>
</div>
</div>
<div id="tabset-2-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-2-tab">
<p>Además, reutilizamos la función previamente empleada para graficar la distribución del genoma, pero esta vez generamos un gráfico que muestra la distribución de las longitudes de todas las secuencias codificantes, correspondientes a cada genoma.</p>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Function plot_violin_length call</span></span>
<span><span class="fu">plot_violin_length</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">all_code_prot_data</span>,</span>
<span>  <span class="st">"genome"</span>, </span>
<span>  plot_title <span class="op">=</span> <span class="st">"coding sequence size"</span>, </span>
<span>  x_label <span class="op">=</span> <span class="st">"Number of base pairs"</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell quarto-layout-panel" data-layout="[100]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="geneFeature_files/figure-html/plotting-length all coding protein-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Secuencias por genoma y largo total de la secuencia.</figcaption></figure>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
<hr></section></section><section id="procesamiento-de-las-secuencias-dgf-1" class="level2" data-number="2.4"><h2 data-number="2.4" class="anchored" data-anchor-id="procesamiento-de-las-secuencias-dgf-1">
<span class="header-section-number">2.4</span> Procesamiento de las secuencias <code>DGF-1</code>
</h2>
<p>La función <code>calculate_fasta_feature</code> se usa para calcular para cada secuencia el contenido de <code>GC</code> y la <code>longitud (pb)</code> de <code>DGF-1</code>. Se crea una <code>tabla resumen</code>, que agrupa las secuencias por cada genoma <code>TriTrypDB-68_TcruziDm28c2018_Genome.fasta, TriTrypDB-68_TcruziBrazilA4_Genome.fasta, TriTrypDB-68_TcruziYC6_Genome.fasta, TcDm25_TcruziTcDm25H1_Genome.fasta</code>, mostrando el <code>DGF-1</code>, así como la <code>longitud</code> mínima, media y máxima, así como el contenido de <code>GC</code> mínimo, medio y máximo.</p>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">target_protein</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"all_annotate_gene_gc_length.tsv"</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Crear una tabla con el conteo de secuencias por archivo</span></span>
<span><span class="va">file_sequence_counts</span> <span class="op">&lt;-</span> <span class="va">target_protein</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">file</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>Total_Sequences <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html">n</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>            min_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">Length</span><span class="op">)</span>,</span>
<span>            mean_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">Length</span><span class="op">)</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>            max_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">Length</span><span class="op">)</span>,</span>
<span>            min_gc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">GC_Content</span><span class="op">)</span>,</span>
<span>            mean_gc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">GC_Content</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>            max_gc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">GC_Content</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Mostrar la tabla de conteo de secuencias</span></span>
<span><span class="fu"><a href="https://glin.github.io/reactable/reference/reactable.html">reactable</a></span><span class="op">(</span><span class="va">file_sequence_counts</span>, highlight <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-f546ff12693ce55767db" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f546ff12693ce55767db">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"file":["TcruziBrazilA4","TcruziDm28c2018","TcruziTcDm25H1","TcruziYC6"],"Total_Sequences":[44,75,192,102],"min_length":[267,7380,228,220],"mean_length":[6021,10334,5944,8811],"max_length":[10500,10821,10551,10521],"min_gc":[0.5528,0.6206,0.5769,0.5882],"mean_gc":[0.634,0.637,0.648,0.643],"max_gc":[0.6977,0.6436,0.691,0.6884]},"columns":[{"id":"file","name":"file","type":"character"},{"id":"Total_Sequences","name":"Total_Sequences","type":"numeric"},{"id":"min_length","name":"min_length","type":"numeric"},{"id":"mean_length","name":"mean_length","type":"numeric"},{"id":"max_length","name":"max_length","type":"numeric"},{"id":"min_gc","name":"min_gc","type":"numeric"},{"id":"mean_gc","name":"mean_gc","type":"numeric"},{"id":"max_gc","name":"max_gc","type":"numeric"}],"highlight":true,"dataKey":"d061efe3b64c136d94ecf45ff33f0d9a"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<section id="visualización-de-la-longitud-y-contenido-de-gc-de-dgf-1" class="level3" data-number="2.4.1"><h3 data-number="2.4.1" class="anchored" data-anchor-id="visualización-de-la-longitud-y-contenido-de-gc-de-dgf-1">
<span class="header-section-number">2.4.1</span> Visualización de la Longitud y Contenido de GC de <code>DGF-1</code>
</h3>
<p>Se generan gráficos que representan el <code>contenido de GC</code> y la <code>longitud</code> de cada secuencia de <code>DGF-1</code>. En esta seccion reutilizaremos parte del codigo escrito llamando a las funciones que crean los plot de <code>GC</code> y de <code>longitud</code> de las secuencias.</p>
<div class="tabset-margin-container"></div><div class="panel-tabset nav-pills">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-3-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-1" role="tab" aria-controls="tabset-3-1" aria-selected="true">GC</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-3-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-3-2" role="tab" aria-controls="tabset-3-2" aria-selected="false">Length</a></li>
</ul>
<div class="tab-content nav-pills">
<div id="tabset-3-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-3-1-tab">
<p>Se usa la funcion <code>plot_violin_gc</code>, dando lugar a un gráfico violín que muestra la distribución del contenido de GC en distintos genomas de las secuencias <code>DGF-1</code>.</p>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">var_title_gc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">key_name</span>,<span class="st">" GC"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Function plot_violin_gc call</span></span>
<span><span class="fu">plot_violin_gc</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">target_protein</span>, </span>
<span>  plot_title <span class="op">=</span> <span class="va">var_title_gc</span>, </span>
<span>  y_label <span class="op">=</span> <span class="st">"GC Content"</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell quarto-layout-panel" data-layout="[100]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="geneFeature_files/figure-html/plotting-gc-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Secuencias por genoma y contenido GC.</figcaption></figure>
</div>
</div>
</div>
</div>
</div>
<div id="tabset-3-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-3-2-tab">
<p>Se usa la función previamente empleada para graficar la distribución del genoma, pero esta vez generamos un gráfico que muestra la distribución de las longitudes de las distintas secuencias de <code>DGF-1</code>, correspondientes a cada genoma.</p>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">var_title</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">key_name</span>,<span class="st">" size"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Function plot_violin_length call</span></span>
<span><span class="fu">plot_violin_length</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">target_protein</span>,</span>
<span>  <span class="st">"gene"</span>, </span>
<span>  plot_title <span class="op">=</span> <span class="va">var_title</span>, </span>
<span>  x_label <span class="op">=</span> <span class="st">"Number of base pairs"</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell quarto-layout-panel" data-layout="[100]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="geneFeature_files/figure-html/plotting-length-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Secuencias por genoma y largo total de la secuencia.</figcaption></figure>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
<hr></section></section><section id="filtrado-de-las-secuencias-de-dgf-1-por-el-largo" class="level2" data-number="2.5"><h2 data-number="2.5" class="anchored" data-anchor-id="filtrado-de-las-secuencias-de-dgf-1-por-el-largo">
<span class="header-section-number">2.5</span> Filtrado de las secuencias de DGF-1 por el largo</h2>
<p>Se filtran las secuencias de DGF-1 de acuerdo a un largo mínimo (9900 pb). Las secuencias menores a este minimo se descartaran para este analisis inicial, y serán estudiadas para determinar si son pseudogenes.</p>
<section id="resumen-de-información-de-dgf-1-completas-en-cada-genoma-filtradas-por-largo" class="level3" data-number="2.5.1"><h3 data-number="2.5.1" class="anchored" data-anchor-id="resumen-de-información-de-dgf-1-completas-en-cada-genoma-filtradas-por-largo">
<span class="header-section-number">2.5.1</span> Resumen de Información de DGF-1 completas en cada genoma (filtradas por largo)</h3>
<p>Se genera una tabla con las estadístias descriptivas de las DGF-1 completas</p>
<div class="cell">
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-78045300a15d5049f8ff" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-78045300a15d5049f8ff">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"file":["TcruziBrazilA4","TcruziDm28c2018","TcruziTcDm25H1","TcruziYC6"],"Total_Sequences":[18,71,67,82],"min_length":[10056,10131,10003,9951],"mean_length":[10388,10412,10399,10401],"max_length":[10500,10821,10551,10521],"min_gc":[0.58,0.6332,0.5799,0.5882],"mean_gc":[0.635,0.638,0.637,0.639],"max_gc":[0.6441,0.6436,0.6431,0.6449]},"columns":[{"id":"file","name":"file","type":"character"},{"id":"Total_Sequences","name":"Total_Sequences","type":"numeric"},{"id":"min_length","name":"min_length","type":"numeric"},{"id":"mean_length","name":"mean_length","type":"numeric"},{"id":"max_length","name":"max_length","type":"numeric"},{"id":"min_gc","name":"min_gc","type":"numeric"},{"id":"mean_gc","name":"mean_gc","type":"numeric"},{"id":"max_gc","name":"max_gc","type":"numeric"}],"highlight":true,"dataKey":"785616028cb5a9584afed187012a0818"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section><section id="visualización-de-la-longitud-y-contenido-de-gc-de-dgf-1-completas" class="level3" data-number="2.5.2"><h3 data-number="2.5.2" class="anchored" data-anchor-id="visualización-de-la-longitud-y-contenido-de-gc-de-dgf-1-completas">
<span class="header-section-number">2.5.2</span> Visualización de la Longitud y Contenido de GC de <code>DGF-1</code> completas</h3>
<p>Se representan el <code>contenido de GC</code> y la <code>longitud</code> de cada secuencia de <code>DGF-1</code> completas (filtrada por un <code>largo minimo</code>).</p>
<div class="tabset-margin-container"></div><div class="panel-tabset nav-pills">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-4-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-4-1" role="tab" aria-controls="tabset-4-1" aria-selected="true">GC</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-4-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-4-2" role="tab" aria-controls="tabset-4-2" aria-selected="false">Length</a></li>
</ul>
<div class="tab-content nav-pills">
<div id="tabset-4-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-4-1-tab">
<p>Se usa la funcion <code>plot_violin_gc</code>, para representar la distribución del contenido de GC en distintos genomas de las secuencias <code>DGF-1</code> filtradas por un largo minimo.</p>
<div>

</div>
<div class="cell quarto-layout-panel" data-layout="[100]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="geneFeature_files/figure-html/plotting-GC-filter-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Secuencias por genoma y largo total de la secuencia.</figcaption></figure>
</div>
</div>
</div>
</div>
</div>
<div id="tabset-4-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-4-2-tab">
<p>Se usa la función <code>plot_violin_length</code> para representar distribución de las secuencias <code>DGF-1</code> filtradas.</p>
<div>

</div>
<div class="cell quarto-layout-panel" data-layout="[100]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="geneFeature_files/figure-html/plotting-length-filter-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Secuencias por genoma y largo total de la secuencia.</figcaption></figure>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
<hr></section></section><section id="resumen-de-información-de-secuencias-anotadas-como-dgf-1-menores-al-largo-minimo-establecido" class="level2" data-number="2.6"><h2 data-number="2.6" class="anchored" data-anchor-id="resumen-de-información-de-secuencias-anotadas-como-dgf-1-menores-al-largo-minimo-establecido">
<span class="header-section-number">2.6</span> Resumen de Información de secuencias anotadas como <code>DGF-1</code> menores al largo minimo establecido</h2>
<p>Se crea una nueva tabla resumen a partir de las secuencias que se descartaran por no cumplir con el minimo de largo establecido.</p>
<div class="cell">
<div class="cell-output-display">
<div class="reactable html-widget html-fill-item" id="htmlwidget-21a84fae6b5624d3b6a8" style="width:auto;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-21a84fae6b5624d3b6a8">{"x":{"tag":{"name":"Reactable","attribs":{"data":{"file":["TcruziBrazilA4","TcruziDm28c2018","TcruziTcDm25H1","TcruziYC6"],"Total_Sequences":[26,4,125,20],"min_length":[267,7380,228,220],"mean_length":[2997,8954,3556,2293],"max_length":[9410,9801,9835,9772],"min_gc":[0.5528,0.6206,0.5769,0.5971],"mean_gc":[0.634,0.63,0.654,0.658],"max_gc":[0.6977,0.6363,0.691,0.6884]},"columns":[{"id":"file","name":"file","type":"character"},{"id":"Total_Sequences","name":"Total_Sequences","type":"numeric"},{"id":"min_length","name":"min_length","type":"numeric"},{"id":"mean_length","name":"mean_length","type":"numeric"},{"id":"max_length","name":"max_length","type":"numeric"},{"id":"min_gc","name":"min_gc","type":"numeric"},{"id":"mean_gc","name":"mean_gc","type":"numeric"},{"id":"max_gc","name":"max_gc","type":"numeric"}],"highlight":true,"dataKey":"969d3d96b504b9594d64dc7cdacec171"},"children":[]},"class":"reactR_markup"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<hr></section><section id="caracterizacion-de-los-nuevos-genes-dgf-1-anotados" class="level2" data-number="2.7"><h2 data-number="2.7" class="anchored" data-anchor-id="caracterizacion-de-los-nuevos-genes-dgf-1-anotados">
<span class="header-section-number">2.7</span> Caracterizacion de los nuevos genes <code>DGF-1</code> anotados</h2>
<div class="cell">
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Load the table generated by the second script</span></span>
<span><span class="va">new_gene_annotate</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"new_gene_length_gc.tsv"</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># all_code_prot_data &lt;- read.csv("../output_directory/all_code_protein_gc_length.csv", sep = ",", header = TRUE)</span></span>
<span></span>
<span><span class="co"># Crear una tabla con el conteo de secuencias por archivo</span></span>
<span><span class="va">new_gene_annotatet_summary</span> <span class="op">&lt;-</span> <span class="va">new_gene_annotate</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">filename</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>Total_Sequences <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html">n</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>            min_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span>,</span>
<span>            mean_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span>, <span class="fl">0</span><span class="op">)</span>,</span>
<span>            max_length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">length</span><span class="op">)</span>,</span>
<span>            min_gc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">gc_content</span><span class="op">)</span>,</span>
<span>            mean_gc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">gc_content</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span>,</span>
<span>            max_gc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">gc_content</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="visualización-de-la-longitud-y-contenido-de-gc-de-el-total-de-los-nuevos-genes-dedgf-1" class="level3" data-number="2.7.1"><h3 data-number="2.7.1" class="anchored" data-anchor-id="visualización-de-la-longitud-y-contenido-de-gc-de-el-total-de-los-nuevos-genes-dedgf-1">
<span class="header-section-number">2.7.1</span> Visualización de la Longitud y Contenido de GC de el total de los nuevos genes de<code>DGF-1</code>
</h3>
<p>Se generan gráficos que representan el <code>contenido de GC</code> y la <code>longitud</code> de cada secuencia de <code>DGF-1</code> re anotadas. En esta seccion reutilizaremos parte del codigo escrito llamando a las funciones que crean los plot de <code>GC</code> y de <code>longitud</code> de las secuencias.</p>
<div class="tabset-margin-container"></div><div class="panel-tabset nav-pills">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-5-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-5-1" role="tab" aria-controls="tabset-5-1" aria-selected="true">GC</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-5-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-5-2" role="tab" aria-controls="tabset-5-2" aria-selected="false">Length</a></li>
</ul>
<div class="tab-content nav-pills">
<div id="tabset-5-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-5-1-tab">
<p>Se usa la funcion <code>plot_violin_gc</code>, dando lugar a un gráfico violín que muestra la distribución del contenido de GC en distintos genomas de las secuencias <code>DGF-1</code>.</p>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb12"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">total_gene_annotate</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"total_gene_length_gc.tsv"</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># total_gene_annotatet_summary &lt;- total_gene_annotate %&gt;%</span></span>
<span><span class="co">#   group_by(file) %&gt;%</span></span>
<span><span class="co">#   summarise(Total_Sequences = n(),</span></span>
<span><span class="co">#             min_length = min(Length),</span></span>
<span><span class="co">#             mean_length = round(mean(Length), 0),</span></span>
<span><span class="co">#             max_length = max(Length),</span></span>
<span><span class="co">#             min_gc = min(GC_Content),</span></span>
<span><span class="co">#             mean_gc = round(mean(GC_Content), 3),</span></span>
<span><span class="co">#             max_gc = max(GC_Content))</span></span>
<span></span>
<span><span class="va">var_title_gc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">key_name</span>,<span class="st">" GC"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Function plot_violin_gc call</span></span>
<span><span class="fu">plot_violin_gc</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">total_gene_annotate</span>, </span>
<span>  plot_title <span class="op">=</span> <span class="va">var_title_gc</span>, </span>
<span>  y_label <span class="op">=</span> <span class="st">"GC Content"</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell quarto-layout-panel" data-layout="[100]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="geneFeature_files/figure-html/plotting-gc re annotate-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Secuencias por genoma y contenido GC re anotadas</figcaption></figure>
</div>
</div>
</div>
</div>
</div>
<div id="tabset-5-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-5-2-tab">
<p>Se usa la función previamente empleada para graficar la distribución del genoma, pero esta vez generamos un gráfico que muestra la distribución de las longitudes de las distintas secuencias de <code>DGF-1</code>, correspondientes a cada genoma.</p>
<details class="code-fold"><summary>Code</summary><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">var_title</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">key_name</span>,<span class="st">" size"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Function plot_violin_length call</span></span>
<span><span class="fu">plot_violin_length</span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">total_gene_annotate</span>,</span>
<span>  <span class="st">"gene"</span>, </span>
<span>  plot_title <span class="op">=</span> <span class="va">var_title</span>, </span>
<span>  x_label <span class="op">=</span> <span class="st">"Number of base pairs"</span></span>
<span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details><div class="cell quarto-layout-panel" data-layout="[100]">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: flex-start;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="geneFeature_files/figure-html/plotting-length re annotate-1.png" class="img-fluid figure-img" width="672"></p>
<figcaption>Secuencias por genoma y largo total de la secuencia re anotadas</figcaption></figure>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


</section></section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./index.html" class="pagination-link" aria-label="Análisis de Secuencias">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Análisis de Secuencias</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./dinucleotideFreq.html" class="pagination-link" aria-label="Nucleotide frequency">
        <span class="nav-page-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Nucleotide frequency</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>